src/api/server.rs

```diff
@@ -64,13 +67,21 @@ fn make_routes(playground: bool) -> BoxedFilter<(impl Reply,)> {
     let not_found = warp::any().and_then(|| async { Err(warp::reject::not_found()) });
 
     // GraphQL query and subscription handler
-    let graphql_handler = warp::path("graphql").and(graphql_subscription(schema.clone()).or(
-        async_graphql_warp::graphql(schema).and_then(
+    let graphql_handler = warp::path("graphql").and(
+        graphql_subscription_with_data(
```

Comment:
In async-graphql, subscriptions and 'regular' queries/mutations have different contexts. The `graphql_subscription_with_data` func can be used to derive context from the initial payload, hence (I assume) the motivation for two functions.

Currently, the `EventInspector` is only provided to subscriptions (i.e. WebSocket connections.) This is to avoid the (marginal) expense of updating the atomic count via an Arc::clone, which would be moot since we have no current use-case for it in queries/mutations.

If we later expand the use-case to include those types (I could see a query that retrieved events over a sampled `interval`, for example), then I'll update the `async_graphql_warp::graphql` call to take an EventInspector, too.